File: frontend/src/service/GoodsService.js
import {GET, POST_V2} from '../utils/Axios';
import {url} from '../config/url';
export const editGood = (data, callback) => {
  const request = `${url}/goods/editGood`;
  POST_V2(request, data, callback);
};

export const addGood = (data, callback) => {
  const request = `${url}/goods/addGood`;
  POST_V2(request, data, callback);
};
export const removeGoods = (data,callback) => {
  const request = `${url}/goods/removeGoods`;
  GET(request, data, callback);
};
export const listGoodsByPage = (data, callback) => {
  const request = `${url}/goods/listGoodsByPage`;
  GET(request, data, callback);
};

export const listMyGoodsByPage = (data, callback) => {
  const request = `${url}/goods/listMyGoodsByPage`;
  GET(request, data, callback);
};

export const findGoodsById = (data, callback) => {
  const request = `${url}/goods/findGoodsById`;
  GET(request, data, callback);
};

//下面这个和上面的区别在于参数，下边只需要goodId，上边那个需要goodsId和userid
export const getGoodsById = (data, callback) => {
  const request = `${url}/goods/getGoodsById`;
  GET(request, data, callback);
};

export const searchGoods = (data, callback) => {
  const request = `${url}/goods/searchGoods`;
  GET(request, data, callback);
};

export const getWanter = (data, callback) => {
  const request = `${url}/goods/getWanter`;
  GET(request, data, callback);
};
export const getMaxGoodsPage = (data, callback) => {
  const request = `${url}/goods/getMaxGoodsPage`;
  GET(request, data, callback);
};

File: backend/src/main/java/com/backend/passthemon/service/GoodsService.java
package com.backend.passthemon.service;

import com.alibaba.fastjson.JSONObject;
import com.backend.passthemon.dto.GoodInfoDto;
import com.backend.passthemon.dto.GoodsDto;
import com.backend.passthemon.entity.Goods;
import org.springframework.data.domain.Page;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;

import java.awt.print.Book;
import java.util.List;
import java.util.Map;

public interface GoodsService {
    JSONObject listGoodsByPages(Integer fetchPage, Integer userId, Integer category, Integer attrition);
    List<GoodsDto> listMyGoodsByPages(Integer fetchPage, Integer userId);
    List<Goods> searchGoods(String search, Integer fetchPage);
    Goods findGoodsById(Integer goodsId);
    Goods addGood(Goods goods);
    Integer editGood(GoodInfoDto goodInfoDto,Integer userId); //成功返回1,如果商品的userId和userId不一致返回-1
    Integer removeGoods(Integer goodsId,Integer userId);  //成功返回1，如果goods的userId和goodsId不相同，返回-1
    void changeState(Integer state,Integer goodsId);
    Integer getMaxGoodsPage(Integer category, Integer attrition);
}
